contracts/access/manager/AccessManager.sol

```diff
@@ -572,15 +572,13 @@ contract AccessManager is Context, Multicall, IAccessManager {
 
         uint48 minWhen = Time.timestamp() + setback;
 
-        if (when == 0) {
-            when = minWhen;
-        }
-
-        // If caller is not authorised, revert
-        if (!immediate && (setback == 0 || when < minWhen)) {
+        // if call is not authorized, or if requested timing is too soon
+        if ((!immediate && setback == 0) || (when > 0 && when < minWhen)) {
             revert AccessManagerUnauthorizedCall(caller, target, bytes4(data[0:4]));
         }
 
+        uint48 timepoint = uint48(Math.max(when, minWhen)); // cast is safe: both inputs are uint48
```

Comment:
The reason I had reused the `when` variable was because the upgradeable contracts were failing with "stack too deep". You can see that it is failing now too.